### ISSUE TITLE
[BUG] mlflow.utils.async_logging.async_logging_queue  too many 500 error responses

### ISSUE THREAD

**Issue by yukiman76** [mlflow/mlflow#14627]
### Issues Policy acknowledgement

- [x] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Where did you encounter this bug?

Local machine

### MLflow version

mlflow==2.20.2
mlflow-skinny==2.20.2

### System information

- Ubuntu 22.04.4 LTS (Jammy Jellyfish)
- Python 3.10.16

### Describe the problem

2025/02/17 15:39:42 ERROR mlflow.utils.async_logging.async_logging_queue: Run Id e0dc683c39c5400f8b2a1b4b20c990e5: Failed to log run data: Exception: API request to http://192.168.1.7:8080/api/2.0/mlflow/runs/log-batch failed with exception HTTPConnectionPool(host='192.168.1.7', port=8080): Max retries exceeded with url: /api/2.0/mlflow/runs/log-batch (Caused by ResponseError('too many 500 error responses'))
2025/02/17 15:39:42 ERROR mlflow.utils.async_logging.async_logging_queue: Run Id e0dc683c39c5400f8b2a1b4b20c990e5: Failed to log run data: Exception: API request to http://192.168.1.7:8080/api/2.0/mlflow/runs/log-batch failed with exception HTTPConnectionPool(host='192.168.1.7', port=8080): Max retries exceeded with url: /api/2.0/mlflow/runs/log-batch (Caused by ResponseError('too many 500 error responses'))
2025/02/17 15:39:44 ERROR mlflow.utils.async_logging.async_logging_queue: Run Id e0dc683c39c5400f8b2a1b4b20c990e5: Failed to log run data: Exception: API request to http://192.168.1.7:8080/api/2.0/mlflow/runs/log-batch failed with exception HTTPConnectionPool(host='192.168.1.7', port=8080): Max retries exceeded with url: /api/2.0/mlflow/runs/log-batch (Caused by ResponseError('too many 500 error responses'))

### Tracking information

```shell
REPLACE_ME
```

### Code to reproduce issue

```
REPLACE_ME
```

### Stack trace

***** Running training *****
  Num examples = 412,519
  Num Epochs = 20
  Instantaneous batch size per device = 1
  Total train batch size (w. parallel, distributed & accumulation) = 64
  Gradient Accumulation steps = 16
  Total optimization steps = 128,900
  Number of trainable parameters = 1,100,048,384
  0%|                                                                                                                 | 0/128900 [00:00<?, ?it/s]2025/02/17 15:39:42 ERROR mlflow.utils.async_logging.async_logging_queue: Run Id e0dc683c39c5400f8b2a1b4b20c990e5: Failed to log run data: Exception: API request to http://192.168.1.7:8080/api/2.0/mlflow/runs/log-batch failed with exception HTTPConnectionPool(host='192.168.1.7', port=8080): Max retries exceeded with url: /api/2.0/mlflow/runs/log-batch (Caused by ResponseError('too many 500 error responses'))
2025/02/17 15:39:42 ERROR mlflow.utils.async_logging.async_logging_queue: Run Id e0dc683c39c5400f8b2a1b4b20c990e5: Failed to log run data: Exception: API request to http://192.168.1.7:8080/api/2.0/mlflow/runs/log-batch failed with exception HTTPConnectionPool(host='192.168.1.7', port=8080): Max retries exceeded with url: /api/2.0/mlflow/runs/log-batch (Caused by ResponseError('too many 500 error responses'))
2025/02/17 15:39:44 ERROR mlflow.utils.async_logging.async_logging_queue: Run Id e0dc683c39c5400f8b2a1b4b20c990e5: Failed to log run data: Exception: API request to http://192.168.1.7:8080/api/2.0/mlflow/runs/log-batch failed with exception HTTPConnectionPool(host='192.168.1.7', port=8080): Max retries exceeded with url: /api/2.0/mlflow/runs/log-batch (Caused by ResponseError('too many 500 error responses'))
^Ad

### Other info / logs

```
REPLACE_ME
```

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by daniellok-db** [comment]
@yukiman76 could you provide a minimal code snippet that reproduces the error?

also if you have the tracking server logs (the thing that is throwing the 500s), please paste those in as well.

**Comment by yukiman76** [comment]
This was using the SFTTrainer with mlflow as an output while in a DDP traning loop, let me see if I can get a basic demo built that would exhibit the behavior

**Comment by daniellok-db** [comment]
@yukiman76 thank you, that would be super helpful

**Comment by yukiman76** [comment]
To follow up on this report, we moved the code base to another sever as a test (with the same versions of all packages) and it does not fail there, we are going to rebuild the server
